cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -64,20 +66,26 @@
   private final ScheduledExecutorService scheduledExecutorService;
   private final Store store;
   private final ProgramRuntimeService runtimeService;
+  private final CConfiguration configuration;
 
   @Inject
-  public ProgramLifecycleService(Store store, ProgramRuntimeService runtimeService) {
+  public ProgramLifecycleService(Store store, ProgramRuntimeService runtimeService, CConfiguration configuration) {
     this.store = store;
     this.runtimeService = runtimeService;
     this.scheduledExecutorService = Executors.newScheduledThreadPool(1);
+    this.configuration = configuration;
   }
 
   @Override
   protected void startUp() throws Exception {
     LOG.info("Starting ProgramLifecycleService");
 
+    long interval = configuration.getLong(Constants.AppFabric.PROGRAM_RUNID_CORRECTOR_INTERVAL_SECONDS);
+    if (interval <= 0) {
+      interval = 300L;
```

Comment:
I could add debug log to make it less noisy but still give use some trackability.